### ISSUE TITLE
The output in the source code is 1x116x5040, but after converting to onnx, the output is 1x116x8400

### ISSUE THREAD

**Issue by fish-kong** [ultralytics/ultralytics#3198]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

The output in the source code is 1x116x5040, but after converting to onnx, the output is 1x116x8400

**Comment by glenn-jocher** [comment]
Hi @fish-kong, thank you for reaching out to us with your question.

The change in output shape you are observing after converting to ONNX can have multiple causes. One possibility is that the conversion process may introduce additional layers or operations to the model, resulting in a different output shape.

To better assist you, could you please provide more details about the steps you followed to convert the model to ONNX? It would be helpful to know which tool or library you used for the conversion and any specific settings or parameters you used.

Additionally, if you could provide the exact architecture of the model and the code snippet you used to obtain the 1x116x5040 output, it would help us in understanding the issue better so that we can provide a more accurate response.

Looking forward to your response so that we can assist you further.

**Comment by TimotheeEsin** [comment]
Hi @glenn-jocher. I downloaded 3 different models (yolov8n.pt; yolov8n-seg.pt; yolov8n-cls.pt) and wanted to study what I get as a result. I took your example code

from ultralytics import YOLO

# Load a model
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)

# Use the model
model.train(data="coco128.yaml", epochs=3)  # train the model
metrics = model.val()  # evaluate model performance on the validation set
results = model("https://ultralytics.com/images/bus.jpg")  # predict on an image
path = model.export(format="onnx")  # export the model to ONNX format

I upload yolov8n.onnx to netron and saw at the output layer that consists of 1x84x8400. How it works? What is it 84 and 8400? Can you understand me please?

**Comment by moonbeam5115** [comment]
Looking all over the internet for a good answer to this.

Did you find anything useful?

**Comment by glenn-jocher** [comment]
@moonbeam5115 hi, the ONNX model output size difference you're noticing may come from added layers or operations during the model conversion process. 

In your case, the output format of "1x84x8400" means that the model outputs a tensor with 1 batch, 84 classes, and 8400 bounding box attributes. The '84' represents the classes that the model can recognize and the '8400' indicates the attributes of the bounding boxes that the model predicts for each class.

I hope this explanation helps to clarify your question. If you require further assistance, please don't hesitate to reach out again.

**Comment by AccSSS** [comment]
@glenn-jocher hi，how to distinguish multiple pose from data，If 84 represents one person's key point, how to obtain it for multiple people?

**Comment by glenn-jocher** [comment]
@AccSSS hello, 

YOLOv8's output tensor is structured such that it holds the bounding box predictions for each detected object. For your specific question about pose estimation and if "84" represents key points of one person, here's a brief explanation:

In the output tensor "1x84x8400", the '84' doesn't denote key points of a single person. Rather, '84' typically denotes the number of classes that the model has been trained to detect. 

If you are working with pose estimation and the keypoints of individuals are included as separate classes, then yes in your context it could mean keypoints for each individual. However, typically in object detection tasks, '84' would represent the 84 different objects that your model can recognize. The latter dimension '8400' signifies the bounding box attributes that the model predicts for each class.

Multiple individuals in an image are distinguished based on different bounding boxes. Each person will have a separate bounding box prediction, each with its own set of key points.

I hope this clarifies your question. If you have further queries, please feel free to ask.
